### PR TITLE
[jasmine-3.1] Add the`saveArgumentsByValue` function

### PR DESCRIPTION
**Pull request by robert-j-webb**
See https://github.com/jasmine/jasmine/blob/15f969bee7066526ba44e562407c728d4a1740df/src/core/CallTracker.js\#L117 for definition

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jasmine/jasmine/blob/15f969bee7066526ba44e562407c728d4a1740df/src/core/CallTracker.js\#L117
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

No, this was just missed. It says that it has been in since version 2.5.0.

- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by chivesrs** [comment]
https://jasmine.github.io/api/3.5/Spy_calls

Code seems fine

Re structure, please also add this to the outer index.d.ts file (in root directory).

ts3.1 directory means TypeScript 3.1, not Jasmine 3.1. This is because TypeScript made significant changes in its 3.1 release regarding type inference, so we have 2 separate definitions, one for older code (looser typings), and one for modern code (stricter typings).

Basically, all changes are added in both files.

**Review by chivesrs — CHANGES_REQUESTED** [review]
See previous comment re all index.d.ts files

**Comment by robert-j-webb** [comment]
Copied the update and tests to Jasmine/v2 and jasmine/index.d.ts

**Comment by chivesrs** [comment]
Looks good. As a final check, can you take the tests you just wrote and run them in plukr/stackblitz/favorite similar tool and link here?

**Review by chivesrs — APPROVED** [review]

**Comment by robert-j-webb** [comment]
Hmm I got it working with 2.5.0, but I had a tiny bug in my code:

https://plnkr.co/edit/xxWSHRDmQJrSQF08?preview

Should I add it to v2 if it only works with v2.5.0+?

**Comment by chivesrs** [comment]
If the plunkr works with the latest release of Jasmine, and is compatible with the typings in both the outer `/` and `ts3.1/` directories, then I am happy with this PR.

According to the header of the `v2` definitions, it is for Jasmine 2.8 (no idea what patch version). If you can get it working with the latest patch of Jasmine 2.8, then add it to the `v2` directory, otherwise just add it to the main 2 directories.

**Review by chivesrs — APPROVED** [review]

**Comment by robert-j-webb** [comment]
Here it is working with 2.8.0 [the only version I found]: https://plnkr.co/edit/ZTd7pxdaEljLGuhn

And here it is working with 3.5.0: https://plnkr.co/edit/bp0VXKGchbj3gb3e

**Comment by chivesrs** [comment]
Looks good. You'll have to wait for a repo maintainer to merge. No estimate on time for that.

**Comment by DanielRosenwasser** [comment]
Thanks for the patience!

**Comment by robert-j-webb** [comment]
No worries! I still have to wait for the typings to be imported to my company anyway haha!

**Pull request merged**
